### PR TITLE
Guest auto recurring tokenization

### PR DESCRIPTION
**Pull request by ScottPowell14**

### CODE REVIEW

**Comment by ScottPowell14** [comment]
Need to review an issue with invalid merchant signature for redirects

**Comment by ScottPowell14** [comment]
> Need to review an issue with invalid merchant signature for redirects

resolved

File: cartridges/int_adyen_overlay/cartridge/scripts/adyenRedirectVerificationSHA256.ds [hunk 1]
```diff
@@ -107,7 +109,19 @@ function verify (args) {
 	var currencyCode:String	 		= order.currencyCode;
 	var merchantReference:String 	= orderNo;
 	args.shopperEmail				= order.customerEmail;
-	args.shopperReference			= currentUser.ID;
+	var customerID = '';
```
Comment (rikterbeek) [inline cartridges/int_adyen_overlay/cartridge/scripts/adyenRedirectVerificationSHA256.ds]: make this null

File: cartridges/int_adyen_overlay/cartridge/scripts/adyenRedirectVerificationSHA256.ds [hunk 2]
```diff
@@ -107,7 +109,19 @@ function verify (args) {
...
+		customerID = currentUser.ID;	
+		
+	} else {
+		// set shopperReference for guest shopper (order.customerNo)
+		if (!empty(order.customerNo)) {
+			customerID = order.customerNo;	
+		}
+	}
+	
+	args.shopperReference			= customerID;
```
Comment (rikterbeek) [inline cartridges/int_adyen_overlay/cartridge/scripts/adyenRedirectVerificationSHA256.ds]: if not null then add this into the args.

File: cartridges/int_adyen_overlay/cartridge/scripts/adyenRedirectVerificationSHA256.ds [hunk 3]
```diff
@@ -116,11 +130,17 @@ function verify (args) {
     var shipBeforeDate : String = new Date();
     shipBeforeDate.setDate(shipBeforeDate.getDate() + 5);
     args.shipBeforeDate = shipBeforeDate.toISOString();
-
-	if (recurringType.value  == ""  || recurringType.value == "null" || recurringType.value == null || !recurringEnabled || !currentUser.registered) {
-		recurringType = "";
-	}	
-	else recurringType = recurringType.value; //recurringType.value has type string 
+    
+    var setRecurringType = "";
```
Comment (rikterbeek) [inline cartridges/int_adyen_overlay/cartridge/scripts/adyenRedirectVerificationSHA256.ds]: setRecurringType make null value.

File: cartridges/int_adyen_overlay/cartridge/scripts/adyenRedirectVerificationSHA256.ds [hunk 4]
```diff
@@ -116,11 +130,17 @@ function verify (args) {
...
     args.shipBeforeDate = shipBeforeDate.toISOString();
-
-	if (recurringType.value  == ""  || recurringType.value == "null" || recurringType.value == null || !recurringEnabled || !currentUser.registered) {
-		recurringType = "";
-	}	
-	else recurringType = recurringType.value; //recurringType.value has type string 
+    
+    var setRecurringType = "";	
+    
+    if (recurringEnabled && currentUser.registered) {
```
Comment (rikterbeek) [inline cartridges/int_adyen_overlay/cartridge/scripts/adyenRedirectVerificationSHA256.ds]: combine this if else statement with the if/else above.

File: cartridges/int_adyen_overlay/cartridge/scripts/adyenRedirectVerificationSHA256.ds (hunk 2 above)
Comment (rikterbeek) [reply cartridges/int_adyen_overlay/cartridge/scripts/adyenRedirectVerificationSHA256.ds]: where do we use this for ?

File: cartridges/int_adyen_overlay/cartridge/scripts/adyenRedirectVerificationSHA256.ds [hunk 5]
```diff
@@ -131,8 +151,8 @@ function verify (args) {
 	adyenRequest.put("skinCode", skinCode );
 	adyenRequest.put("merchantAccount", merchantAccount );
 	adyenRequest.put("shopperEmail", order.customerEmail );
-	adyenRequest.put("shopperReference", currentUser.ID );
-	adyenRequest.put("recurringContract", recurringType );
+	adyenRequest.put("shopperReference", customerID ); // currentUser.ID
```
Comment (rikterbeek) [inline cartridges/int_adyen_overlay/cartridge/scripts/adyenRedirectVerificationSHA256.ds]: if customId is null don't add this line same

File: cartridges/int_adyen_overlay/cartridge/scripts/adyenRedirectVerificationSHA256.ds [hunk 6]
```diff
@@ -131,8 +151,8 @@ function verify (args) {
...
+	adyenRequest.put("recurringContract", setRecurringType );
```
Comment (rikterbeek) [inline cartridges/int_adyen_overlay/cartridge/scripts/adyenRedirectVerificationSHA256.ds]: if setRecurringType is null don't add this line same

**Pull request merged**